### DISCUSSION TITLE
ics-query - what happens when in ics files

### DISCUSSION THREAD

**Discussion by niccokunzmann** [collective/icalendar#706] [Show and tell]
Using `icalendar`, we can parse .ics files well. I made a new tool that allows us to find out what happens when: `ics-query`.

[Check out ics-query on PyPI](https://pypi.org/project/ics-query/)

## Example

Installation:

```shell
pip install ics-query
```

You can easily get a calendar from the web and see what is on. In this example, we show which German National Holidays happen in August 2024:

```shell
$ wget -qO- 'https://www.calendarlabs.com/ical-calendar/ics/46/Germany_Holidays.ics' | ./ics-query at 2024-08 - -
BEGIN:VEVENT
SUMMARY:Assumption Day (BY\, SL)
DTSTART;VALUE=DATE:20240815
DTEND;VALUE=DATE:20240815
DTSTAMP:20231013T092513Z
UID:65290cf9326601697189113@calendarlabs.com
SEQUENCE:0
DESCRIPTION:Visit https://calendarlabs.com/holidays/us/the-assumption-of-m
 ary.php to know more about Assumption Day (BY\, SL). \n\n Like us on Faceb
 ook: http://fb.com/calendarlabs to get updates
LOCATION:Germany
STATUS:CONFIRMED
TRANSP:TRANSPARENT
END:VEVENT
```

## Future

I was thinking that e.g. the #702 jCal proposed by @abe-101 would be really cool with this one as you can also use [jq](https://jqlang.github.io/jq/) to query the result, then.

It will be really easy with this to create small tools and automations which work with icalendar in the background.

## Feedback

Feedback is very welcome!

**Comment by abe-101** [comment]
I like that idea
I see how it can be useful in automation scripts?

**Reply by niccokunzmann** [reply]
For automation, I think one can query a calendar and see what events have happened until now and if they were not used yet, one can run a script based on that.
